index.bs

```diff
@@ -2032,6 +2075,8 @@ The <dfn method for="XRWebGLLayer">getViewport(|view|)</dfn> method, when invoke
   1. If |session| is not equal to |layer|'s [=XRWebGLLayer/session=], throw an {{InvalidStateError}} and abort these steps.
   1. If |frame|'s [=XRFrame/active=] boolean is `false`, throw an {{InvalidStateError}} and abort these steps.
   1. If |view|'s [=XRView/frame time=] is not equal to |frame|'s [=XRFrame/time=], throw an {{InvalidStateError}} and abort these steps.
+  1. If |view|'s [=view/viewport scale=] was changed, [=update the viewports=] for |session|.
```

Comment:
This step is a little awkward, as it may allow for running the `update the viewports` algorithm (which updates ALL viewports) up to once per viewport if you change and then fetch each viewport in the course of a frame.

Also, the language "was changed" is a little loose. I'd kind of prefer an explicit dirty flag or similar.

Reply:
I've updated it to do a direct replacement in the list of viewport objects, and changed the "obtain a scaled viewport" method to return an `XRViewport` instead of `glViewport` so that it can be shared more easily with the "update the viewports" method.

The "was changed" is now a direct comparison of "current viewport scale" and "requested viewport scale", and I've moved the "viewport modifiable" check and update here.